tests/test_alphanet.rs

```diff
@@ -26,50 +103,124 @@ fn run_multiple_nodes(num_nodes: usize, num_trials: usize) {
             boot_nodes,
             chain_spec.clone(),
         );
-        boot_nodes = vec![node.node_addr()];
-        node.start();
         nodes.push(node);
     }
-    //        thread::sleep(Duration::from_secs(10));
+    (init_balance, account_names, nodes)
+}
+
+const CHAIN_SPEC_FILE: &str = "tmp/alphanet_chain.json";
+
+fn start_test_node_process(num_nodes: usize, node_index: usize) -> Child {
+    let account_name = format!("near.{}", node_index);
+    let keygen_path = format!("test{}/storage/keystore/", node_index);
+    let mut keygen_command = Command::new("cargo");
+    keygen_command.args(&[
+        "run",
+        "--package", "keystore",
+        "--",
+        "keygen",
+        "--test-seed", account_name.as_str(),
+        "-p", keygen_path.as_str(),
+    ]);
+    let node_infos: Vec<String> = (0..num_nodes)
+        .map(|i|
+           format!("127.0.0.1:{}/{}", 5000 + i, get_peer_id_from_seed(i as u32 + 1))
+        ).collect();
+    let boot_node = node_infos[(node_index + num_nodes - 1) % num_nodes].as_str();
+    let pubkey = InMemorySigner::from_seed(&account_name, &account_name).public_key;
+
+    let mut start_node_command = Command::new("cargo");
+    start_node_command.args(&[
+        "run",
+        "--",
+        "--rpc_port", format!("{}", 3030 + node_index).as_str(),
+        "--addr", format!("127.0.0.1:{}", 3000 + node_index).as_str(),
+        "--base-path", format!("test{}", node_index).as_str(),
+        "--test-network-key-seed", format!("{}", node_index + 1).as_str(),
+        "--chain-spec-file", CHAIN_SPEC_FILE,
+        "--boot-nodes", boot_node,
+        "-a", account_name.as_str(),
+        "-k", format!("{}", pubkey).as_str(),
+    ]);
+    keygen_command.spawn().expect("keygen command failed").wait().expect("keygen command failed");
+    start_node_command.spawn().expect("start_node command failed")
+}
+
+// DISCLAIMER. These tests are very heavy and somehow manage to interfere with each other.
+// If you add multiple tests and they start failing consider splitting it into several *.rs files
+// to ensure they are not run in parallel.
+
+#[test]
```

Comment:
I recommend extracting this test into a separate file so that it does not run in parallel with other heavy tests which currently causes CPU starvation and timeout, see https://github.com/nearprotocol/nearcore/issues/718

Reply:
You would have to introduce a `test-utils` crate for that. CC @bowenwang1996 
Relevant discussion: https://github.com/nearprotocol/nearcore/pull/717